cnf-tests/docgen/cmd/generate.go

```diff
@@ -0,0 +1,197 @@
+}
+
+var testListTemplate = `
+<!--- IMPORTANT!
+This file is generated manually. To add a new description please run run
+hack/fill-empty-docs.sh, check the json description files and fill the missing descriptions (the placeholder is XXXXXX)
+--->
+
+# Validation Test List
+
+The validation tests are preliminary tests intended to verify that the instrumented features are available on the cluster.
+| Test Name | Description |
+| -- | ----------- |{{ with .ValidationList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+# CNF Tests List
+The cnf tests instrument each different feature required by CNF. Following, a detailed description for each test.
+
+## SCTP
+
+| Test Name | Description |
+| -- | ----------- |{{ with .SCTPList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+## SR-IOV
+
+| Test Name | Description |
+| -- | ----------- |{{ with .SRIOVList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+## Performance
+
+| Test Name | Description |
+| -- | ----------- |{{ with .PerformanceList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+## PTP
+
+| Test Name | Description |
+| -- | ----------- |{{ with .PTPList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+## DPDK
+
+| Test Name | Description |
+| -- | ----------- |{{ with .DPDKList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+
+## Others
+
+| Test Name | Description |
+| -- | ----------- |{{ with .OthersList }}{{ range . }}
+| {{ .Name }} | {{ .Description }} | {{ end }}{{ end }}
+`
+
+// generateCmd represents the generate command
+var generateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		updateTestList(markdownFile, e2eDescriptions, validationDescriptions)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(generateCmd)
+	generateCmd.Flags().StringVar(&markdownFile, "target", "", "The path of the markdown file")
+	generateCmd.Flags().StringVar(&e2eDescriptions, "testsjson", "", "The json file containing the descriptions of the e2e tests")
+	generateCmd.Flags().StringVar(&validationDescriptions, "validationjson", "", "The json file containing the descriptions of the validation tests")
+	cobra.MarkFlagRequired(generateCmd.LocalFlags(), "target")
+	cobra.MarkFlagRequired(generateCmd.LocalFlags(), "testsjson")
+	cobra.MarkFlagRequired(generateCmd.LocalFlags(), "validationjson")
+}
+
+type TestDescription struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type TemplateData struct {
```

Comment:
can this be just a list of `[]TestDescription` then extend the `TestDescription` struct to something like

```golang
suiteName string
tests []TestDescription
```

This way we can make the template much smaller, just have two for loop one over suites and inside one over the tests

Reply:
Makes sense, need to keep the validation separate though as the template is slightly different with a different description there.